client/json-rpc/rpc_procedures.c

```diff
@@ -549,6 +553,17 @@ cJSON * method_xdag_do_xfer(struct xdag_rpc_context * ctx, cJSON *params, cJSON
 			if (cJSON_IsString(json_address)) {
 				strcpy(address, json_address->valuestring);
 			}
+
+			cJSON *json_remark = cJSON_GetObjectItem(param, "remark");
+			if (cJSON_IsString(json_remark)) {
+				if(strlen(json_remark->valuestring) < 32) {
+					strcpy(remark, json_remark->valuestring);
+				} else {
+					ctx->error_code = 1;
+					ctx->error_message = strdup("Transacion remark exceed max length 32.");
```

Comment:
is there the free somewhere?
if strdup fail, so if ctx->error_message == NULL, what will happen?